packages/theming/theme-tokens/src/pipeline-input/token-input-black.json

```diff
@@ -0,0 +1,218 @@
+{
+  "$schema": "https://raw.githubusercontent.com/microsoft/fluentui-token-pipeline/main/docs/token.schema.json",
+  "Meta": {
+    "FluentUITokensVersion": 0
+  },
+  "Set": {
+    "NeutralForeground2Brand": {
+      "Fill": {
+        "Color": {
+          "Hover": { "aliasOf": "Global.Color.Brand.Tint.40" },
+          "Pressed": { "aliasOf": "Global.Color.Brand.Tint.10" }
```

Comment:
Trying to understand what "Hover" and "Pressed" mean here for "NeutralForeground2Brand". If I set this alias token as the foreground color of a control, does this mean it will automatically change colors depending on the state (like hover and pressed) and I don't need to explicitly set a different color for hover and pressed states in my token/settings file?

Reply:
Oh nevermind, I think I see how it works. These are eventually split out into "neutralForeground2Hover" and "neutralForeground2Pressed" and that value for those changes based on the theme (black, dark, colorful, etc.)?

Reply:
Unfortunately no, although that would be nice.

Design gave us the alias tokens as "neutralForeground2Brand", "neutralForeground2BrandHover", neutralForeground2BrandPressed", etc. This file is the file that gets fed into the pipeline, which expects the input to be formatted as:

```
<Root of alias token name>: {
    Fill: {
        Color: {
            Rest/Hover/Pressed/etc. as separate entries
        }
    }
}
```

This particular file is only overriding entries that are different from token-input-dark.json (the web dark mode version). The pipeline allows you to stack inputs, similar to how styles work in RN. That's why some entries are missing.

There's a separate place that takes care of mapping the output json structure to actual alias tokens stored on the Theme object, which was moved to mapPipelineToTheme. These convert the output to the alias token names given by design, so that when you're given redlines, you can reference those theme object entries to fill out token entries. Design will specify whether they want hover/pressed, or not.

Reply:
Ah okay 🙂 thanks for the explanation!